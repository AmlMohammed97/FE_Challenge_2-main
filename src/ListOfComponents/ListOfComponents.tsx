import React from 'react';

/**
 *
 *
 * @param {number} columns - list columns number
 * @param {JSX.Element} listOfComponents - list of components to be rendered
 *
 */
type Props = {
  columns: number;
  listOfComponents: JSX.Element[];
};

const ListOfComponents: React.FunctionComponent<Props> = ({ columns, listOfComponents }): JSX.Element => {
  return (
    <div data-testid="ListOfComponents" style={{ columnCount: columns }}>
      {listOfComponents}
    </div>
  );
};

export default ListOfComponents;
