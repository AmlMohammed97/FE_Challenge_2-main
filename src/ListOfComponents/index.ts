export { default } from './ListOfComponents';
export * from './ListOfComponents';
