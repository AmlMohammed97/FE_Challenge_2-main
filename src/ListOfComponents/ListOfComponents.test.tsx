import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import ListOfComponents from '.';

describe('ListOfComponents', () => {
  describe('snapshot', () => {
    it('renders correctly', () => {
      const listOfComponents = renderer.create(<ListOfComponents columns={3} listOfComponents={[<div key={1} />, <div key={2} />]} />).toJSON();
      expect(listOfComponents).toMatchSnapshot();
    });
  });
  describe('initialize', () => {
    it('renders with three columns', () => {
      render(<ListOfComponents columns={3} listOfComponents={[]} />);
      const listOfComponents = screen.getByTestId('ListOfComponents');
      expect(listOfComponents).toHaveStyle({ 'column-count': '3' });
    });
    it('renders with list of two elements', () => {
      render(<ListOfComponents columns={1} listOfComponents={[<div key={1} />, <div key={2} />]} />);
      const listOfComponents = screen.getByTestId('ListOfComponents');
      expect(listOfComponents.children.length).toEqual(2);
    });
  });
});
