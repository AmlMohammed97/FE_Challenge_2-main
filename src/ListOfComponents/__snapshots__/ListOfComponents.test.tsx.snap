// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfComponents snapshot renders correctly 1`] = `
<div
  data-testid="ListOfComponents"
  style={
    {
      "columnCount": 3,
    }
  }
>
  <div />
  <div />
</div>
`;
